### PR TITLE
Adding Pipeline to Repo

### LINKED ISSUE #29: Create CI/CD pipeline for App
I want to use our tests to ensure that our code is always up to standard and that a pipeline that checks our code will be a step in doing this.

Acceptance Criteria:

- [x] Pipeline uses GitHub actions 
- [x] On each code push the pipeline is run before any merging can occur
- [x] Front and Backend tests are run in the pipeline
- [x] Pipeline is triggered automatically on Pull request or push to the main branch
- [x] Merging is blocked if any test fails
- [x] Pipeline Incorporates static code analysis tool (e.g ESLint) to detect coding errors, inconsistencies, and potential vulnerabilities
- [x] Pipeline is extensible to add deployment jobs at a later time

### PR DESCRIPTION
**Pull request by JackMcPhillips1543**
Fixes #29 

- Creating CI/CD pipeline for repo
- Starts up virtual env
- Run frontend and backend test
- Lints frontend and backend 

Why was the change needed:
- Needed a pipeline
- Needed to implement tests
- Needed a way to check the quality of the code

How was it changed:
- Added .yml file
- downloaded pylint, and npm lint
- moved test files to one place

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        run: |
+          cd ${{github.workspace}}/frontend
+          npm install
+          npm test
+
+      - name: Run backend tests
+        run: |
+          source env/bin/activate
+          cd ${{github.workspace}}/backend
+          pytest -s websocketUnitTest.py
```
Comment (Nissa9902) [inline .github/workflows/main.yml:48]: Great test for WebSocket! Check edge cases too

File: frontend/src/App.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ function App() {
 
     websocket.current.onclose = (event) => {
       console.log("WebSocket connection closed: ", event);
-      websocketConnected = false;
+      // websocketConnected = false;
     };
 
     return () => {
```
Comment (Nissa9902) [inline frontend/src/App.js:40]: Proper state management for WebSocket connection in frontend.

File: requirements.txt [hunk 3]
```diff
@@ -12,4 +12,9 @@ pytest==8.3.3
...
 httpx==0.27.2
 pytest-asyncio==0.24.0
-websockets==13.1
\ No newline at end of file
+websockets==13.1
+astroid==3.3.5 
+dill==0.3.9
+isort==5.13.2
+pylint==3.3.1
+tomlkit==0.13.2
```
Comment (Nissa9902) [inline requirements.txt:20]: Good job setting up the CI workflow for both building and linting! The matrix and Python setup is clear and structured well.

**Review by Nissa9902 — APPROVED** [review]

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
```
Comment (Nissa9902) [inline .github/workflows/main.yml:49]: Great test for WebSocket! Check edge cases too

**Review by Nissa9902 — APPROVED** [review]

**Review by git-voo — APPROVED** [review]
Great job!
The follow up task would be to refactor the codes in our codebase to pass linter tests and ensure the pipeline runs successfully. But that's an independent issue outside the scope of your PR
